tests/osgi-tests/src/test/java/org/wso2/carbon/osgi/config/EnvironmentUtils.java

```diff
@@ -0,0 +1,97 @@
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.osgi.config;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Utility class for setting environment variables for test cases.
+ *
+ * @since 1.0.0
+ */
+public class EnvironmentUtils {
+    private static Logger logger = LoggerFactory.getLogger(EnvironmentUtils.class);
+
+    private static final String PROCESS_ENVIRONMENT = "java.lang.ProcessEnvironment";
+    private static final String THE_ENVIRONMENT_FILED = "theEnvironment";
+    private static final String THE_CASE_INSENSITIVE_ENVIRONMENT = "theCaseInsensitiveEnvironment";
+    private static final String COLLECTIONS_UNMODIFIABLE_MAP = "java.util.Collections$UnmodifiableMap";
+    private static final String FIELD_M = "m";
+
+    /**
+     * Set environment variable for a given key and value.
+     *
+     * @param key Environment variable key.
+     * @param value Environment variable value.
+     */
+    public static void setEnv(String key, String value) {
+        Map<String, String> newenv = new HashMap<>();
+        newenv.put(key, value);
+        setEnv(newenv);
+    }
+
+    /**
+     * Set environment variable from given map.
+     *
+     * @param newVariables Map of variables to put into environment variables.
+     */
+    public static void setEnv(Map<String, String> newVariables) {
+        Map<String, String> newenv = new HashMap<>();
+        newenv.putAll(System.getenv());
+        newenv.putAll(newVariables);
+        try {
+            Class<?> processEnvironmentClass = Class.forName(PROCESS_ENVIRONMENT);
+
+            Field theEnvironmentField = processEnvironmentClass.getDeclaredField(THE_ENVIRONMENT_FILED);
+            theEnvironmentField.setAccessible(true);
+            Map<String, String> env = (Map<String, String>) theEnvironmentField.get(null);
+            env.putAll(newenv);
+
+            Field theCaseInsensitiveEnvironmentField = processEnvironmentClass
+                    .getDeclaredField(THE_CASE_INSENSITIVE_ENVIRONMENT);
+            theCaseInsensitiveEnvironmentField.setAccessible(true);
+            Map<String, String> cienv = (Map<String, String>) theCaseInsensitiveEnvironmentField.get(null);
+            cienv.putAll(newenv);
+        } catch (NoSuchFieldException e) {
+            Class[] classes = Collections.class.getDeclaredClasses();
+            Map<String, String> env = System.getenv();
+            Arrays.asList(classes).stream().filter(cl -> COLLECTIONS_UNMODIFIABLE_MAP.equals(cl.getName())).forEach(
+                    (cl) -> {
+                        try {
+                            Field field = cl.getDeclaredField(FIELD_M);
+                            field.setAccessible(true);
+                            Object obj = field.get(env);
```

Comment:
This method set environment variables to the current process using reflections. If ProcessEnvironment class doesn't have theEnvironment or theCaseInsensitiveEnvironment field, execution comes to this catch clause. In the catch clause, we set the new environment variables to a field called "m". You can refer [1] for more details.
[1] http://stackoverflow.com/questions/318239/how-do-i-set-environment-variables-from-java